crates/net/network/src/transactions/mod.rs

```diff
@@ -1618,11 +1620,12 @@ mod tests {
             .disable_discovery()
             .listener_port(0)
             .build(client);
+        let transactions_manager_config = config.transactions_manager_config.clone();
```

Comment:
you don't need to clone this, unnecessary. you should move it into `transactions` not pass by ref, since it won't be used anymore after that.

Reply:
emm. wait what i'm confused, the config is consumed when construct NetworkManager::new(config) how do you move the field out of it?

Reply:
by addressing this first https://github.com/paradigmxyz/reth/pull/6594/files#r1489759678, this problem here should be solved as a side effect.

Reply:
well here, since the config implements default, you can move the config out of the network config like this
```rust
std::mem::take(&mut config.transactions_manager_config)
```
<ins>but</ins>, someone in the future may, with good reason, assume that field will continue being the actual transactions manager config during the whole program, and create a logic bug based on that assumption if it is replaced with the default. because of this we would better use `Option<TransactionsManagerConfig>` for the field in that case. however, the constructor for the `NetworkManager` and `TransactionsManager` will only be called once, so the clone here is actually trivial for performance. it's just good practice to avoid them when possible.

Reply:
yeah, or can take a ref as previously done, so seem current code is fine?

Reply:
yepp it's good